docs: add awesome-faker link in guide -> community (#2625)
faker-js/faker: docs/guide/index.md

## Community

If you have questions or need help, reach out to the community via [Discord](https://chat.fakerjs.dev) and [GitHub Discussions](https://github.com/faker-js/faker/discussions).

> You can see awesome things people in the community have built with Faker at [Awesome Faker](https://github.com/faker-js/awesome-faker).

Reviewer: This list also includes things that are not strictly build "with" faker. I suggest using the wording from the awesome repo

```suggestion
You can see awesome things related to the Faker community at [Awesome Faker](https://github.com/faker-js/awesome-faker).
```
